### PR TITLE
Verify Clean Up ODS Deployer Post-Migration

### PR DESCRIPTION
**Pull request by pablofelix**
Signed-off-by: Pablo Felix Estevez Pico <pestevez@redhat.com>

### CODE REVIEW

File: tests/Tests/100__deploy/100__installation/109__operator.robot [hunk 1]
```diff
@@ -47,6 +47,60 @@ Verify That The Operator Pod Does Not Get Stuck After Upgrade
...
         END
     END
+Verify Clean Up ODS Deployer Post-Migration
+    [Tags]    Tier1
+    ...       ODS-1767
+    ${version_check} =    Is RHODS Version Greater Or Equal Than    1.17.0
+    IF    ${version_check} == False
+        Log    Skipping test case as RHODS version is less than 1.17.0
+        Skip
+    END
```
Comment (tarukumar) [inline tests/Tests/100__deploy/100__installation/109__operator.robot:57]: There is a Keyword which can  reduce the number of line  to single one . `Skip If RHODS Version Greater Or Equal Than    version=1.17.0`

**Comment by tarukumar** [comment]
LGTM..apart from above comment

**Review by bdattoma — DISMISSED** [review]

File: tests/Tests/100__deploy/100__installation/109__operator.robot (hunk 1 above)
Comment (jgarciao) [reply tests/Tests/100__deploy/100__installation/109__operator.robot:57]: In this case it should skip if version < 1.17.

@pablofelix you could create a keyword `Skip If RHODS Version Less Or Equal Than` in tests/Resources/Common.robot ?

File: tests/Tests/100__deploy/100__installation/109__operator.robot [hunk 2]
```diff
@@ -47,6 +47,60 @@ Verify That The Operator Pod Does Not Get Stuck After Upgrade
             Log Error And Fail Pods When Pods Were Terminated    ${operator_pod_info}    Opertator Pod Stuck
         END
     END
+Verify Clean Up ODS Deployer Post-Migration
+    [Tags]    Tier1
```
Comment (jgarciao) [inline tests/Tests/100__deploy/100__installation/109__operator.robot:51]: Missing [Documentation]

**Review by jgarciao — CHANGES_REQUESTED** [review]

**Comment by jgarciao** [comment]
@pablofelix Could you link the jenkins job  where this was tested?

File: tests/Tests/100__deploy/100__installation/109__operator.robot [hunk 3]
```diff
@@ -47,6 +47,60 @@ Verify That The Operator Pod Does Not Get Stuck After Upgrade
...
+    ${version_check} =    Is RHODS Version Greater Or Equal Than    1.17.0
+    IF    ${version_check} == False
+        Log    Skipping test case as RHODS version is less than 1.17.0
+        Skip
+    END
+    ${cro_pod}=    Run Keyword And Return Status
+    ...    Oc Get    kind=Pod    api_version=v1    label_selector=name=cloud-resource-operator
+    IF    ${cro_pod} == True
+        Fail    CRO pod found after migration
+    END
```
Comment (lugi0) [inline tests/Tests/100__deploy/100__installation/109__operator.robot:62]: Failing here means the rest of the TC isn't executed.
Would it be more useful to keep going and collect all potential failures other than the presence of the CRO?

File: tests/Tests/100__deploy/100__installation/109__operator.robot (hunk 1 above)
Comment (tarukumar) [reply tests/Tests/100__deploy/100__installation/109__operator.robot:57]: yes correct

**Comment by pablofelix** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-test-pr/23/robot/

File: tests/Tests/100__deploy/100__installation/109__operator.robot (hunk 3 above)
Comment (jgarciao) [reply tests/Tests/100__deploy/100__installation/109__operator.robot:64]: Maybe `Run Keyword And Continue On Failure     Verify CRO Is Not Deployed` could have worked better. Also, we have the keyword  `Check If POD Exists` in Robots.robot that could have been used.

I'm going to merge this PR anyway as it is urgent

**Review by jgarciao — APPROVED** [review]

**Review by jgarciao — APPROVED** [review]

**Review by bdattoma — APPROVED** [review]

**Pull request merged**